ci/functional/test_main.sh

```diff
@@ -2,6 +2,34 @@
 
 set -eux
 
+report_junit() {
+    local name="$1"
+    local results="$2"
+    local nodes="$3"
+
+    clush -o '-i ci_key' -l root -w "$nodes" --rcopy "$results"
+
+    local num_reports="$(ls "$results".* | wc -l)"
+    if [ "$num_reports" = "0" ]; then
+        echo "No results found to report as JUnit results"
+        ls -l
+        return
+    fi
+
+    mkdir -p "$STAGE_NAME"/framework/
+
+    cat <<EOF > "$STAGE_NAME"/framework/framework_results.xml
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite errors="$num_reports" failures="0" name="$name" skipped="0"
+           tests="$num_reports" time="0" timestamp="$(date +%FT%T)">
+$(cat $results.*)
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]

Reply:
Inside a heredoc.